frontend/packages/console-plugin-sdk/src/registry.ts

```diff
@@ -43,34 +42,35 @@ export class ExtensionRegistry {
     this.extensions = _.flatMap(plugins.map((p) => p.extensions));
   }
 
-  public get<E extends Extension>(typeGuard: ExtensionTypeGuard<E>): E[] {
-    return this.extensions.filter(typeGuard);
-  }
-
-  public getRequiredFlags(typeGuards: ExtensionTypeGuard<ExtensionWithFlags>[]) {
+  public getGatingFlagNames(typeGuards: ExtensionTypeGuard<ExtensionWithFlags>[]) {
     return _.flatMap(typeGuards.map((tg) => this.extensions.filter(tg)))
-      .filter((e) => e.properties.required)
+      .filter((e) => e.properties.required || e.properties.disallowed)
       .reduce(
-        (requiredFlags, e) => _.uniq([...requiredFlags, ..._.castArray(e.properties.required)]),
+        (gatingFlags, e) =>
+          _.uniq([
+            ...gatingFlags,
+            ..._.castArray(e.properties.required),
+            ..._.castArray(e.properties.disallowed),
+          ]),
```

Comment:
```
.reduce( (gatingFlags, ext) => ext.properties.required || ext.properties.disallowed ? 
    _.uniq(....)
: gatingFlags)
```
maybe only use reduce?

Reply:
Sure, I can put the `filter` logic inside the `reduce` handler.